instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/OtelResourceAutoConfiguration.java

```diff
@@ -42,27 +43,16 @@ public ResourceProvider otelEnvironmentResourceProvider() {
   }
 
   @Bean
-  public ResourceProvider otelSpringResourceProvider() {
-    return new SpringResourceProvider();
+  public ResourceProvider otelSpringResourceProvider(
+      @Autowired(required = false) BuildProperties buildProperties) {
```

Comment:
I think you can use `Optional<BuildProperties>` as the method argument to let spring inject an `Optional`